controllers/actions.github.com/ephemeralrunner_controller.go

```diff
@@ -236,6 +267,29 @@ func (r *EphemeralRunnerReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	}
 }
 
+func (r *EphemeralRunnerReconciler) cleanupRunnerFromService(ctx context.Context, ephemeralRunner *v1alpha1.EphemeralRunner, log logr.Logger) (ctrl.Result, error) {
+	err := r.deleteRunnerFromService(ctx, ephemeralRunner, log)
+	if err == nil { // if NO error
+		err := patch(ctx, r.Client, ephemeralRunner, func(obj *v1alpha1.EphemeralRunner) {
+			controllerutil.RemoveFinalizer(obj, ephemeralRunnerActionsFinalizerName)
+		})
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
+
+	actionsError := &actions.ActionsError{}
+	if errors.As(err, &actionsError) &&
+		actionsError.StatusCode == http.StatusBadRequest &&
+		strings.Contains(actionsError.ExceptionName, "JobStillRunningException") {
+		log.Info("Runner is still running the job. Re-queue in 30 seconds")
+		return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
+	}
+	log.Error(err, "Failed clean up runner from the service")
+	return ctrl.Result{}, err
```

Comment:
```suggestion
	actionsError := &actions.ActionsError{}
	err := r.deleteRunnerFromService(ctx, ephemeralRunner, log)
	if err != nill {
	   if errors.As(err, &actionsError) &&
	      actionsError.StatusCode == http.StatusBadRequest &&
	      strings.Contains(actionsError.ExceptionName, "JobStillRunningException") {
	        log.Info("Runner is still running the job. Re-queue in 30 seconds")
		return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
	   }
	   
	   log.Error(err, "Failed clean up runner from the service")
           return ctrl.Result{}, err
	}
        
       	log.Info("Successfully removed runner registration from service")
	err := patch(ctx, r.Client, ephemeralRunner, func(obj *v1alpha1.EphemeralRunner) {
		controllerutil.RemoveFinalizer(obj, ephemeralRunnerActionsFinalizerName)
	})
	if err != nil {
		return ctrl.Result{}, err
	}
	
	log.Info("Successfully removed runner registration finalizer")
	return ctrl.Result{}, nil
```

You might want to fix the format a little bit. 😄